### PR TITLE
[ONNX] Don't download ONNX model by default

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

This PR makes sure that ONNX weights are only ever by default downloaded by ONNX pipelines, but not any PyTorch pipelines.

The user is also given the option to have more fine-grained control via a `use_onnx` download argument.

### CODE REVIEW

File: src/diffusers/pipelines/pipeline_utils.py [hunk 1]
```diff
@@ -1411,6 +1412,9 @@ def download(cls, pretrained_model_name, **kwargs) -> Union[str, os.PathLike]:
             ):
                 ignore_patterns = ["*.bin", "*.msgpack"]
 
+                if not use_onnx:
```
Comment (echarlaix) [inline src/diffusers/pipelines/pipeline_utils.py:1415]: Should this be set by default for all the `OnnxStableDiffusionXxxPipeline` ?

**Comment by JacquiML** [comment]
Hey @sayakpaul , in `from_pretrained` func, when calling `cls.download`, should we as `use_onnx` as an input parameter?
current:
```
 if not os.path.isdir(pretrained_model_name_or_path):
            cached_folder = cls.download(
                pretrained_model_name_or_path,
                cache_dir=cache_dir,
                resume_download=resume_download,
                force_download=force_download,
                proxies=proxies,
                local_files_only=local_files_only,
                use_auth_token=use_auth_token,
                revision=revision,
                from_flax=from_flax,
                use_safetensors=use_safetensors,
                custom_pipeline=custom_pipeline,
                custom_revision=custom_revision,
                variant=variant,
                load_connected_pipeline=load_connected_pipeline,
                **kwargs,
            )
```
proposed (added **use_onnx=use_onnx,**):
```
 if not os.path.isdir(pretrained_model_name_or_path):
            cached_folder = cls.download(
                pretrained_model_name_or_path,
                cache_dir=cache_dir,
                resume_download=resume_download,
                force_download=force_download,
                proxies=proxies,
                local_files_only=local_files_only,
                use_auth_token=use_auth_token,
                revision=revision,
                from_flax=from_flax,
                use_safetensors=use_safetensors,
                use_onnx=use_onnx,
                custom_pipeline=custom_pipeline,
                custom_revision=custom_revision,
                variant=variant,
                load_connected_pipeline=load_connected_pipeline,
                **kwargs,
            )
```

**Comment by sayakpaul** [comment]
@patrickvonplaten ^.

**Comment by patrickvonplaten** [comment]
Hey @JacquiML,

Good point! Would you like to open a PR for this?

**Comment by JacquiML** [comment]
> Hey @JacquiML,
> 
> Good point! Would you like to open a PR for this?

sure :)

**Pull request merged**